[RAM] Add simplified rules table to connector flyout

JiaweiWu opened the pull request:
## Summary
Resolves: https://github.com/elastic/response-ops-team/issues/174

Adds a rules table to the edit connector flyout view to let the user see what rules are associated with said connector. The user can click on a rule name to bring them to the rules detail page. 

![connector_rules_list](https://github.com/elastic/kibana/assets/74562234/edfe3b39-712f-423c-8ee0-89dbbe34e7bd)

### Checklist
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

mdefazio reviewed (changes requested):
@JiaweiWu A few thoughts on this:

### Tabs:
I think the tab order should be:
1. Configuration
2. Rules
3. Test

### Rule name
Let's present the rule name the same as we do in the Rule table, with the rule type below the rule name, instead of another column: 
**Rulename**
_Rule type_

I think the rule names are long long enough that we will want the extra column width.

### Table
Let's default to 25 for the page size in the pagination.  We want this to be our default for all tables moving forward.

Probably makes sense to add a width to the 'Last response' column.  Not sure what we have on the rules table, but could mimic that here.  Again, trying to get more width for the rule name column. 

Let me know if any of these don't make sense or want to discuss further.

JiaweiWu commented:
@ymao1 Good idea, perhaps a blurb like "We are only displaying rules that you have access to" or something, @mdefazio what do you think?.  Since it's using the same auth mechanism as the rules list, so rules are just filtered out if the user doesn't have access to them.

Also how can I test the case with preconfigured connectors?

ymao1 commented:
> Also how can I test the case with preconfigured connectors?

I have a preconfigured server log in my `kibana.yml`

```
xpack.actions.preconfigured:
  preconfigured-server-log:
    name: 'server log preconfigured'
    actionTypeId: '.server-log'
```

When I create a rule using this connector and look at the connector flyout, I don't see the rule listed (as a superuser)

JiaweiWu commented:
Interesting, so im using `has_reference` to query for these rules, so perhaps the preconfigured rules are not added as references to rule SOs?

ymao1 commented:
That's correct. References only store references to other saved objects. Since preconfigured connectors are not saved objects, they won't count as references.

mdefazio commented:
> Good idea, perhaps a blurb like "We are only displaying rules that you have access to" or something...
My gut reaction says this feels odd—as in, of course we're only displaying rules that I have access to.  But the scenario @ymao1 described makes me think is just a blurb when I try to delete the connector.  Seems like that's the only time I care, right?

JiaweiWu commented:
I fixed the issue @ymao1 discovered WRT preconfigured and system actions. 

I also addressed the comments @mdefazio suggested.

![new_connector](https://github.com/elastic/kibana/assets/74562234/48cd0bdd-c537-4312-b2aa-804032d6bd61)

Let me know what we decide on rule auth for connectors

mdefazio reviewed (approved):
Thank you! LGTM!

JiaweiWu commented:
@elasticmachine merge upstream

XavierM commented on x-pack/plugins/alerting/common/action_ref_prefix.ts:
```diff
@@ -0,0 +1,12 @@
+export const preconfiguredConnectorActionRefPrefix = 'preconfigured:';
+
+// NOTE: Changing this prefix will require a migration to update the prefix in all existing `rule` saved objects
+export const systemConnectorActionRefPrefix = 'system_action:';
```
where do we have `system_action`?

XavierM replied:
I just realized that we merged some code around system action already. So all good here

XavierM commented on x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/connector_rules_list.tsx:
```diff
@@ -0,0 +1,242 @@
+      };
+    }
+
+    if (connector.isSystemAction) {
```
@cnasikas can we talk about this ?

XavierM commented on x-pack/plugins/triggers_actions_ui/public/types.ts:
```diff
@@ -802,14 +803,14 @@ export interface ConnectorServices {
 }
 
 export interface RulesListFilters {
```
do we really want to put everything as conditional attributes? It feels weird!

JiaweiWu replied:
Its actually odd that `RulesListFilters` wasn't all conditional since the `loadRulesWithKueryFilter` has all of its params marked as optional. 

If I don't use optional here then in https://github.com/elastic/kibana/pull/176456/files#diff-8cd0ce22105ab63d0b2a2d5c45c4afe34eb2c95fbda327092dfabb049d23cff0R78, we need to have bunch of empty arrays for no reason.

XavierM commented:
I am wondering, if we should add the tags in this view since we can search for them in the search bar.Maybe we can update placeholder by `Search rules by name and tags`

<img width="790" alt="image" src="https://github.com/elastic/kibana/assets/189600/d8bf3995-7b80-476c-a68b-e281b25cfceb">

JiaweiWu commented:
@elasticmachine merge upstream

JiaweiWu commented:
@XavierM I address your comments

![new_connector_final_for_sure](https://github.com/elastic/kibana/assets/74562234/8dde9ed6-1de5-4225-9474-96f920ecd68a)
